### PR TITLE
memdocstore: provide an option for a different revision field

### PR DESCRIPTION
**Pull request by jba**
See #2154.

### CODE REVIEW

File: internal/docstore/memdocstore/mem.go [hunk 1]
```diff
@@ -91,35 +94,40 @@ func OpenCollection(keyField string, _ *Options) (*docstore.Collection, error) {
...
 	return docstore.NewCollection(c), nil
 }
 
-func newCollection(keyField string, keyFunc func(docstore.Document) interface{}) (driver.Collection, error) {
+func newCollection(keyField string, keyFunc func(docstore.Document) interface{}, opts *Options) (driver.Collection, error) {
 	if keyField == "" && keyFunc == nil {
 		return nil, gcerr.Newf(gcerr.InvalidArgument, nil, "must provide either keyField or keyFunc")
 	}
+	if opts == nil {
+		opts = &Options{RevisionField: docstore.RevisionField}
```
Comment (vangent) [inline internal/docstore/memdocstore/mem.go]: I think you should create `&Options{}`, then check if `opts.RevisionField == ""` to set the default, in cases someone passes in a zero `Options`.

File: internal/docstore/memdocstore/mem.go [hunk 2]
```diff
@@ -73,14 +73,17 @@ func (*URLOpener) OpenCollectionURL(ctx context.Context, u *url.URL) (*docstore.
 }
 
 // Options are optional arguments to the OpenCollection functions.
-type Options struct{}
+type Options struct {
+	// The name of the field holding the document revision.
```
Comment (vangent) [inline internal/docstore/memdocstore/mem.go]: Add "Defaults to docstore.RevisionField."

File: internal/docstore/memdocstore/mem.go (hunk 1 above)
Comment (jba) [reply internal/docstore/memdocstore/mem.go]: Done.

File: internal/docstore/memdocstore/mem.go (hunk 2 above)
Comment (jba) [reply internal/docstore/memdocstore/mem.go]: done.

**Pull request merged**